### PR TITLE
Enable connection limiting for free tier

### PR DESCRIPTION
**Pull request by Furisto**
## Description
This instructs ws-manager to limit the rate of network connections for non paying customers. Note that this currently only audits (i.e. it records packets that would have been dropped but does not actually drop them). We will use this to refine the limits.

## Related Issue(s)
Fixes https://github.com/gitpod-io/gitpod/issues/13041

## How to test
- User on free tier
  - Make yourself admin in the preview environment
  - Ensure that you are on the free open source plan
  - Start a workspace
  - The workspace should have the `gitpod.io/netConnLimitPerMinute` annotation
  - The other parts of this have already been covered by [other PRs](https://github.com/gitpod-io/gitpod/issues?q=label%3Afeature%3Aconnection-limiting+is%3Aclosed). If you want to be sure, check the logs of ws-daemon (it should contain `will limit network connections`. You can also ssh into the node and then enter the network namespace of the pod with `nsenter -t pid -n`. nft list ruleset should show you a bunch of rules in that namespace.

- Paying customer
  - Ensure that you are on the professional open source plan
  - Start a workspace
  - The workspace should **not** have the `gitpod.io/netConnLimitPerMinute` annotation
  - The other parts should be missing as well

## Release Notes

```release-note
None
```

## Werft options:

- [x] /werft with-preview

### CODE REVIEW

**Comment by roboquat** [comment]
@Furisto: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/server/src/workspace/workspace-starter.ts
Comment (geropl) [inline components/server/src/workspace/workspace-starter.ts:863]: nit: I think it's worth splitting this up into a cascade for readability:
```
               const wsConnectionLimitingEnabled = await getExperimentsClientForBackend().getValueAsync("workspace_connection_limiting", false, {...});
               if (wsConnectionLimitingEnabled) {
                 const shouldLimitNetworkConnections = await this.entitlementService.limitNetworkConnections(user, new Date());
                 if (shouldLimitNetworkConnections) {
                   featureFlags = featureFlags.concat(["workspace_connection_limiting"]);
                 }
               }
```

**Comment by Furisto** [comment]
@geropl PTAL

File: components/server/src/workspace/workspace-starter.ts
Comment (geropl) [inline components/server/src/workspace/workspace-starter.ts:865]: Sorry for the additional turn-around, but: I think It make sense to also sent `teams`. We already fetch it below (line 878) so we can just pull it up. :+1:

File: components/server/src/workspace/workspace-starter.ts
Comment (geropl) [reply components/server/src/workspace/workspace-starter.ts]: Nice.

**Review by geropl — APPROVED** [review]
Tested and works as advertised :heavy_check_mark: 

/hold because I think it's worth addressing [this comment](https://github.com/gitpod-io/gitpod/pull/12585#discussion_r973150360), but leave it up to you to decide.

**Pull request merged**
